targets/TARGET_STM/TARGET_STM32L4/serial_device.c

```diff
@@ -45,69 +45,79 @@ UART_HandleTypeDef uart_handlers[UART_NUM];
 
 static uart_irq_handler irq_handler;
 
+// Defined in serial_api.c
+inline int8_t get_uart_index(UARTName uart_name);
+
 /******************************************************************************
  * INTERRUPTS HANDLING
  ******************************************************************************/
 
-static void uart_irq(int id)
+static void uart_irq(UARTName uart_name)
 {
-    UART_HandleTypeDef * huart = &uart_handlers[id];
-    
-    if (serial_irq_ids[id] != 0) {
-        if (__HAL_UART_GET_FLAG(huart, UART_FLAG_TXE) != RESET) {
-            if (__HAL_UART_GET_IT(huart, UART_IT_TXE) != RESET) {
-                irq_handler(serial_irq_ids[id], TxIrq);
+    int8_t id = get_uart_index(uart_name);
+
+    if (id >= 0) {
```

Comment:
what happens if uart_name is out of defined BASES ? I can see that get uart index returns -1 if not there, but this is not handled here, can it happen?

Reply:
If we have well written all the usart available in the deice it should not happen... Before this it was worse because in some case we could access out the the arrays (uart_handlers[id] and serial_irq_ids[id]).

Reply:
An assert has been added in commit https://github.com/ARMmbed/mbed-os/pull/5962/commits/b6efdd58c8351421fc3b7be3378c647f3e223ae5 to check this index.